pkg/expression/util.go

```diff
@@ -1450,6 +1450,53 @@ func ContainCorrelatedColumn(exprs []Expression) bool {
 	return false
 }
 
+// ProjectionBenefitsFromPushedDown evaluates if the expressions can improve performance when pushed down to TiKV
+// Projections are not pushed down to tikv by default, thus we need to check strictly here to avoid potential performance degradation.
+func ProjectionBenefitsFromPushedDown(exprs []Expression, inputSchemaLen int) bool {
+	allColRef := true
+	colRefCount := 0
+	for _, expr := range exprs {
+		switch v := expr.(type) {
+		case *Column:
+			if v.VirtualExpr != nil {
+				return false
+			}
+			colRefCount = colRefCount + 1
+		case *ScalarFunction:
+			allColRef = false
+			switch v.FuncName.L {
+			case ast.JSONDepth, ast.JSONLength, ast.JSONType, ast.JSONValid, ast.JSONContains, ast.JSONContainsPath,
+				ast.JSONExtract, ast.JSONKeys, ast.JSONSearch, ast.JSONMemberOf, ast.JSONOverlaps:
+				continue
+			case ast.JSONUnquote:
+				arg0 := v.GetArgs()[0]
+				// Only `->>` which parsed to JSONUnquote(CAST(JSONExtract() AS string)) can be pushed down to tikv
+				if fChild, ok := arg0.(*ScalarFunction); ok {
+					if fChild.FuncName.L != ast.Cast {
+						return false
+					}
+					if fGrand, ok := fChild.GetArgs()[0].(*ScalarFunction); ok {
+						if fGrand.FuncName.L != ast.JSONExtract {
+							return false
+						}
```

Comment:
```suggestion
						}
						continue
```

Reply:
It seems like a "continue" is missing.

Reply:
Indeed not straightforward coding here, and missed some situation. Refact it.